src/rtc.rs

```diff
@@ -0,0 +1,639 @@
+    ) -> Self {
+        let mut prec = prec.reset().enable();
+
+        let rcc = unsafe { &*RCC::ptr() };
+
+        // Check and configure clock source
+        let ker_ck = match clock_source {
+            RtcClock::Lse { bypass, freq, .. } => {
+                // Ensure LSE is on and stable
+                rcc.bdcr.modify(|_, w| w.lseon().on().lsebyp().bit(bypass));
+                while rcc.bdcr.read().lserdy().is_not_ready() {}
+
+                Some(freq)
+            }
+            RtcClock::Hse { divider } => {
+                // Set HSE divider
+                assert!(divider < 64, "HSE Divider larger than 63");
+                rcc.cfgr.modify(|_, w| w.rtcpre().bits(divider));
+
+                clocks.hse_ck().map(|x| Hertz(x.0 / u32(divider)))
+            }
+            RtcClock::Lsi => clocks.lsi_ck(),
+        }
+        .expect("rtc_ker_ck not running")
+        .0;
+
+        if ker_ck > (1 << 22) {
+            panic!("rtc_ker_ck too fast for prescaler");
+        }
+
+        // Select RTC kernel clock
+        prec.kernel_clk_mux(match clock_source {
+            RtcClock::Hse { .. } => backup::RtcClkSel::HSE,
+            RtcClock::Lsi => backup::RtcClkSel::LSI,
+            RtcClock::Lse { .. } => backup::RtcClkSel::LSE,
+        });
+
+        // Now we can enable CSS, if required
+        if let RtcClock::Lse { css: true, .. } = clock_source {
+            rcc.bdcr.modify(|_, w| w.lsecsson().security_on());
+        }
+
+        // Disable RTC register write protection
+        rtc.wpr.write(|w| w.key().bits(0xCA));
+        rtc.wpr.write(|w| w.key().bits(0x53));
+
+        // Enter initialization mode
+        rtc.isr.modify(|_, w| w.init().init_mode());
+        while rtc.isr.read().initf().is_not_allowed() {}
+
+        // Configure prescaler for 1Hz clock
+        // Want to maximize a_pre_max for power reasons, though it reduces the
+        // subsecond precision.
+        let total_div = ker_ck;
+        let a_pre_max = 1 << 7;
+        let s_pre_max = 1 << 15;
+
+        let (a_pre, s_pre) = if total_div <= a_pre_max {
+            (total_div, 1)
+        } else if total_div % a_pre_max == 0 {
+            (a_pre_max, total_div / a_pre_max)
+        } else {
+            let mut a_pre = a_pre_max;
+            while a_pre > 1 {
+                if total_div % a_pre == 0 {
+                    break;
+                }
+                a_pre -= 1;
+            }
+            let s_pre = total_div / a_pre;
+            (a_pre, s_pre)
+        };
+
+        rtc.prer.write(|w|
+            w.prediv_s()
+                .bits(u16(s_pre - 1).unwrap())
+                .prediv_a()
+                .bits(u8(a_pre - 1).unwrap())
+        );
+
+        // Exit initialization mode
+        rtc.isr.modify(|_, w| w.init().free_running_mode());
+
+        Rtc { reg: rtc, prec }
+    }
+
+    /// Reads the value of a 32-bit backup register
+    ///
+    /// **Panics:** if reg is greater than 31
```

Comment:
Nice, really good to document these edge cases. There is actually a [standard markdown format](https://rust-lang.github.io/api-guidelines/documentation.html#function-docs-include-error-panic-and-safety-considerations-c-failure) for this, which is slightly different. Maybe worth using that for consistency?